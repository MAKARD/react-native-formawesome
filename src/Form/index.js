// @flow
export * from "./Form";
export * from "./FormProps";
