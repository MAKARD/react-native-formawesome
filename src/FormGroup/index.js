// @flow
export * from "./FormGroup";
export * from "./FormGroupProps";
