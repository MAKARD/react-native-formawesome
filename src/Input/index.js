export * from "./Input";
export * from "./InputProps";
