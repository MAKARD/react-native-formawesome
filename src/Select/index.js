// @flow
export * from "./Select";
export * from "./SelectProps";
