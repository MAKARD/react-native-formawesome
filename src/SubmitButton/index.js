export * from "./SubmitButton";
export * from "./SubmitButtonProps";
