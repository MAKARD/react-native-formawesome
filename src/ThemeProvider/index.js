// @flow
export * from "./ThemeProvider";
export * from "./ThemeProviderProps";
