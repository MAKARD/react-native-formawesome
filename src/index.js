// @flow
export * from "./Form"
export * from "./Input";
export * from "./Select";
export * from "./FormGroup";
export * from "./SubmitButton";
