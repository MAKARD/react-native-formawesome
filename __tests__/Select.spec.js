/* eslint-disable no-undef */
import React from "react";
import { View } from "react-native";
import renderer from "react-test-renderer";
import { ModelValidator } from "react-formawesome-core";

import { Form, FormGroup, Select } from "../src";

class Model {
    name = undefined;
}

describe("Select", () => {
    test("Should render correctly", () => {
        const tree = renderer.create(
          <Form onSubmit={() => undefined} validator={new ModelValidator(Model)}>
              <FormGroup attribute="name">
                  <Select options={[{value: "test", label: "test"}]} />
              </FormGroup>
          </Form>
        ).toJSON();
      
        expect(tree).toMatchSnapshot();
      });
      
      test("Should call onValueChange event if it passed", () => {
          let onChangeTriggered = false;
        
          const tree = renderer.create(
              <Form onSubmit={() => undefined} validator={new ModelValidator(Model)}>
                  <FormGroup attribute="name">
                      <Select 
                          options={[{value: "test", label: "test"}]}
                          onValueChange={() => onChangeTriggered = true}
                      />
                  </FormGroup>
              </Form>
          );
      
          tree.root.findByType(Select).instance.getHandleChange({onChange: () => {}})();
          expect(onChangeTriggered).toBeTruthy();
      });
      
      test("Should apply error styles if error exist", () => {
          const tree = renderer.create(
              <Form onSubmit={() => undefined} validator={new ModelValidator(Model)}>
                  <FormGroup attribute="name">
                      <Select 
                          options={[{value: "test", label: "test"}]}
                          onErrorStyles={{backgroundColor: "#fff"}}
                          style={{paddingLeft: 0}}
                      />
                  </FormGroup>
              </Form>
          );
      
      
          expect(tree.root.findByType(Select).instance.getStyle({error: "error"}))
              .toMatchObject({
                  backgroundColor: "#fff",
                  paddingLeft: 0,
              });
      });
});
