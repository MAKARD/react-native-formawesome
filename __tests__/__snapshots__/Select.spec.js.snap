// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select Should render correctly 1`] = `
<View>
  <View>
    <View>
      <RCTPicker
        items={
          Array [
            Object {
              "label": "test",
              "textColor": undefined,
              "value": "test",
            },
          ]
        }
        onChange={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        selectedIndex={0}
        style={
          Array [
            Object {
              "height": 216,
            },
            undefined,
          ]
        }
      />
    </View>
  </View>
</View>
`;
